server/src/main/java/org/elasticsearch/cluster/block/ClusterBlockException.java

```diff
@@ -110,13 +110,23 @@ private static String buildMessageForIndexBlocks(Map<String, Set<ClusterBlock>>
     @Override
     public RestStatus status() {
         RestStatus status = null;
+        boolean onlyRetryableBlocks = true;
         for (ClusterBlock block : blocks) {
-            if (status == null) {
-                status = block.status();
-            } else if (status.getStatus() < block.status().getStatus()) {
-                status = block.status();
+            boolean isRetryableBlock = block.status() == RestStatus.TOO_MANY_REQUESTS;
+            if (!isRetryableBlock) {
```

Comment:
We have a preference for
```suggestion
            if (isRetryableBlock == false) {
```
to make the negation more visible.